Document/0x04c-Tampering-and-Reverse-Engineering.md

```diff
@@ -196,22 +197,31 @@ Even though most professional GUI-based disassemblers feature scripting faciliti
 
-Typically, symbolic execution is combined with other techniques such as dynamic execution to mitigate the path explosion problem specific to classical symbolic execution. This combination of concrete (actual) and symbolic execution is referred to as *concolic execution* (the name concolic stems from *conc*rete and symb*olic*. Together with improved SMT solvers and current hardware speeds, concolic execution allows to explore paths in medium-size software modules (i.e., on the order of 10s KLOC). However, it also comes in handy for supporting de-obfuscation tasks, such as simplifying control flow graphs. For example, Jonathan Salwan and Romain Thomas have [shown how to reverse engineer VM-based software protections using Dynamic Symbolic Execution](https://triton.quarkslab.com/files/csaw2016-sos-rthomas-jsalwan.pdf "Jonathan Salwan and Romain Thomas: How Triton can help to reverse virtual machine based software protections") (i.e., using a mix of actual execution traces, simulation, and symbolic execution).
+In simple words, symbolic execution is mathematically analyzing a program without executing it. During analysis, each unknown input is represented as a mathematical variable (a symbolic value), and all the operations hence performed on these unknowns is recorded in a form of mathematical equations. These mathematical equations are called `constraints`. In the end of this analysis, a final mathematical equation is obtained, in which the variables are the inputs whose values are not known. SMT solvers are special programs which solve these equations to give possible values for these variables. To visualise this, lets use a function which takes one input and multiplies it by 279 and then subtracts the output by 30. The  mathematical equation for this operation on input will be `(x * 279 - 30)`, where `x` is the variable representing the input to the function. If the value to be multiplied is another input to this function, the equation will become `(x * y - 30)`, where `y` being the another input introduced. We can see the equation is more complex than before. Moving forward, instead of subtracting a constant 30, a value from a global variable is used. This adds another variable to the equation, and makes it `(x * y - z)`. As is the case for global variables, their value can be changed from outside this function, which may lead to different outputs whenever this function is executed. This adds to additional complexity in determining correct solution.
```

Comment:
>In simple words, 

I would just omit this. These words aren't very simple, and this doesn't really add anything.

> Symbolic execution explores the execution tree on a systematic manner

This should be "in a systematic manner", and it could use a comma at the end.

Reply:
@cpholguera My original intention was to keep it absolutely simple. This content is good too, gives enough details on underlying working. And do agree with @Sjord, to remove "in simple words" :)

Reply:
I agree with that. I've updated the suggestion. Thank you both.